src/components/slider.js

```diff
@@ -0,0 +1,371 @@
+import React from "react";
+
+const PREVIOUS = "previous";
+const NEXT = "next";
+
+export const HORIZONTAL = "horizontal";
+export const VERTICAL = "vertical";
+
+const DEFAULT_CLASSNAMES = {
+  previousButton: "previousButton",
+  nextButton: "nextButton",
+  buttonDisabled: "disabled",
+  track: "track",
+  slide: "slide",
+  hidden: "hidden",
+  previous: "previous",
+  current: "current",
+  next: "next",
+  animateIn: "animateIn",
+  animateOut: "animateOut"
+};
+const DEFAULT_DURATION = 2000;
+
+const arrowTransforms = {
+  up: "rotate(90 10 15)",
+  down: "rotate(270 10 15)",
+  left: "rotate(180 10 15)",
+  right: "rotate(0 10 15)"
+};
+function Arrow({ direction = "right" }) {
+  return (
+    <svg
+      xmlns="http://www.w3.org/2000/svg"
+      width="20"
+      height="30"
+      viewBox="0 0 20 30"
+    >
+      <polygon
+        fill="#000"
+        points="20 15 4.228 0 0 3.626 11.954 15 0 26.374 4.228 30"
+        transform={arrowTransforms[direction]}
+      />
+    </svg>
+  );
+}
+
+class Slider extends React.PureComponent {
+  constructor(props) {
+    super(props);
+    const { slideIndex = 0, direction = HORIZONTAL } = this.props;
+    this.state = {
+      currentSlideIndex: slideIndex,
+      animating: false
+    };
+    this.slideCount = React.Children.count(this.props.children);
+    this.direction = direction;
+    this.swipeProperty = direction === HORIZONTAL ? "left" : "top";
+    this.swipeEventProperty = direction === HORIZONTAL ? "clientX" : "clientY";
+  }
+
+  componentDidMount() {
+    this.setupAutoplay();
+  }
+
+  componentWillUnmount() {
+    this.stopAutoplay();
+    if (this.animationTimerId) {
+      clearTimeout(this.animationTimerId);
+    }
+  }
+
+  componentWillReceiveProps(props) {
+    this.slideCount = React.Children.count(props.children);
+    if (this.state.currentSlideIndex >= this.slideCount) {
+      this.setState({ currentSlideIndex: 0 });
+    }
+  }
+
+  setupAutoplay = () => {
+    if (this.props.autoplay && !this.isMouseOver) {
+      this.stopAutoplay();
+      this.autoplayTimerId = setInterval(
+        this.next,
+        parseInt(this.props.autoplay, 10)
+      );
+    }
+  };
+
+  stopAutoplay = () => {
+    if (this.autoplayTimerId) {
+      clearInterval(this.autoplayTimerId);
+    }
+  };
+
+  onAnimationEnd = () => {
+    this.setState({
+      currentSlideIndex: this.nextSlideIndex,
+      animating: false,
+      animation: undefined
+    });
+    this.setupAutoplay();
+    if (typeof this.props.onSlideChange === "function") {
+      this.props.onSlideChange({ slideIndex: this.nextSlideIndex });
+    }
+  };
+
+  isDisabled = () =>
```

Comment:
Generally, its considered best practice to include curly braces when using multi-line.